wdl-doc/src/callable/task.rs

```diff
@@ -0,0 +1,148 @@
+//! Create HTML documentation for WDL tasks.
+
+use maud::Markup;
+use maud::html;
+use wdl_ast::AstToken;
+use wdl_ast::v1::InputSection;
+use wdl_ast::v1::MetadataSection;
+use wdl_ast::v1::OutputSection;
+use wdl_ast::v1::ParameterMetadataSection;
+use wdl_ast::v1::RuntimeSection;
+
+use super::*;
+use crate::parameter::Parameter;
+
+/// A task in a WDL document.
+#[derive(Debug)]
+pub struct Task {
+    /// The name of the task.
+    name: String,
+    /// The meta of the task.
+    meta: MetaMap,
+    /// The input parameters of the task.
+    inputs: Vec<Parameter>,
+    /// The output parameters of the task.
+    outputs: Vec<Parameter>,
+    /// The runtime section of the task.
+    runtime_section: Option<RuntimeSection>,
+}
+
+impl Task {
+    /// Create a new task.
+    pub fn new(
+        name: String,
+        meta_section: Option<MetadataSection>,
+        parameter_meta: Option<ParameterMetadataSection>,
+        input_section: Option<InputSection>,
+        output_section: Option<OutputSection>,
+        runtime_section: Option<RuntimeSection>,
+    ) -> Self {
+        let meta = if let Some(ref mds) = meta_section {
+            parse_meta(mds)
+        } else {
+            MetaMap::default()
+        };
```

Comment:
Nit for consistency since the other `if let` expressions don't use `ref`:

```suggestion
        let meta = if let Some(mds) = meta_section {
            parse_meta(&mds)
        } else {
            MetaMap::default()
        };
```